src/main/java/com/auth0/jwk/Util.java

```diff
@@ -10,4 +15,24 @@ static void checkArgument(boolean arg, String message) {
             throw new IllegalArgumentException(String.valueOf(message));
         }
     }
+
+    private static Map<TimeUnit, ChronoUnit> chronoByTimes = new HashMap<>();
```

Comment:
all these changes could disappear if we simplify the TimeUnit<>Duration conversion